yb-voyager/src/queryparser/unsupportedConstructs.go

```diff
@@ -160,3 +166,147 @@ func containsAdvisoryLocksInNodeList(nodes []*pg_query.Node) bool {
+from_clause:{range_var:{relname:"employees"  inh:true  relpersistence:"p"  location:29}}  where_clause:{a_expr:{kind:AEXPR_OP  name:{string:{sval:"="}}
+lexpr:{column_ref:{fields:{string:{sval:"xmin"}}  location:45}}  rexpr:{param_ref:{number:1  location:52}}  location:50}}
+limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}  alias:{aliasname:"version_info"}}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+
+Query2: SELECT * FROM (SELECT xmin, xmax FROM employees) AS version_info
+ParseTree: version:160001  stmts:{stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{a_star:{}}  location:7}}  location:7}}
+from_clause:{range_subselect:{subquery:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"xmin"}}  location:22}}  location:22}}
+target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"xmax"}}  location:28}}  location:28}}
+from_clause:{range_var:{relname:"employees"  inh:true  relpersistence:"p"  location:38}}
+limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}  alias:{aliasname:"version_info"}}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsSystemColumnsInFromClause(fromClause []*pg_query.Node) bool {
+	for _, fromItem := range fromClause {
+		if subselectNode, isSubselect := fromItem.Node.(*pg_query.Node_RangeSubselect); isSubselect {
+			subSelectStmt := subselectNode.RangeSubselect.Subquery.GetSelectStmt()
+			if subSelectStmt != nil {
+				// Recursively check for advisory locks in the subquery's target list and FROM clause
+				if containsSystemColumnsInTargetList(subSelectStmt.TargetList) {
+					return true
+				}
+				if containsSystemColumnsInFromClause(subSelectStmt.FromClause) {
+					return true
+				}
+				if containsSystemColumnsInWhereClause(subSelectStmt.WhereClause) {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+/*
+Query: SELECT * FROM employees WHERE xmin = $1
+ParseTree: version:160001  stmts:{stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{a_star:{}}  location:7}}  location:7}}
+from_clause:{range_var:{relname:"employees"  inh:true  relpersistence:"p"  location:14}}
+where_clause:{a_expr:{kind:AEXPR_OP  name:{string:{sval:"="}}
+lexpr:{column_ref:{fields:{string:{sval:"xmin"}}  location:30}}  rexpr:{param_ref:{number:1  location:37}}  location:35}}
+limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsSystemColumnsInWhereClause(whereClause *pg_query.Node) bool {
+	if whereClause == nil {
+		return false
+	}
+	lexpr := whereClause.GetAExpr().GetLexpr()
+	columnRef := lexpr.GetColumnRef()
+	if columnRef == nil {
+		return false
+	}
+
+	fields := columnRef.GetFields()
+	if len(fields) == 0 {
+		return false
+	}
+
+	lastField := fields[len(fields)-1]
+	if lastField == nil || lastField.GetString_() == nil {
+		return false
+	}
+
+	colName := lastField.GetString_().Sval
```

Comment:
you can separate this part as a separate function to `getColName(columnRef)` as it used in other functions as well.